sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -115,9 +120,35 @@ abstract class Expression extends TreeNode[Expression] {
     }
   }
 
+  /**
+   * Returns the input variables to this expression.
+   */
+  private def findInputVars(ctx: CodegenContext, eval: ExprCode): Seq[ExprInputVar] = {
+    if (ctx.currentVars != null) {
+      val boundRefs = this.collect {
+        case b @ BoundReference(ordinal, _, _) if ctx.currentVars(ordinal) != null => (ordinal, b)
```

Comment:
shall we add an assert here to guarantee that, `ctx.currentVars(ordinal).code` is empty?

Reply:
It is not necessarily empty. We will also record it if its `code` is not empty.